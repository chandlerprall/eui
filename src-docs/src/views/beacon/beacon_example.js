import React from 'react';

import { renderToHtml } from '../../services';

import { GuideSectionTypes } from '../../components';

import { EuiBeacon } from '../../../../src/components';

import Beacon from './beacon';
const beaconSource = require('!!raw-loader!./beacon');
const beaconHtml = renderToHtml(Beacon);

export const BeaconExample = {
  title: 'Beacon',
  sections: [
    {
      source: [
        {
          type: GuideSectionTypes.JS,
          code: beaconSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: beaconHtml,
        },
      ],
      text: (
        <div>
          <p>Blink, blink</p>
        </div>
      ),
      props: { EuiBeacon },
      demo: <Beacon />,
    },
  ],
};
