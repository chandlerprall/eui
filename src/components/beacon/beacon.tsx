import React, { FunctionComponent, HTMLAttributes } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';

export type EuiBeaconProps = Omit<HTMLAttributes<HTMLDivElement>, 'children'> &
  CommonProps & {
    /**
     * Height and width
     */
    size?: number;
  };

export const EuiBeacon: FunctionComponent<EuiBeaconProps> = ({
  className,
  size = 12,
  style,
  ...rest
}) => {
  const classes = classNames('euiBeacon', className);

  const styles = {
    ...style,
    height: size,
    width: size,
  };

  return <div className={classes} style={styles} {...rest} />;
};
