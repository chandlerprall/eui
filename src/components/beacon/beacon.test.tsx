import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiBeacon } from './beacon';

describe('EuiBeacon', () => {
  test('is rendered', () => {
    const component = render(<EuiBeacon {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });
});
