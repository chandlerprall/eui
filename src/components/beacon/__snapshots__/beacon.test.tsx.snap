// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiBeacon is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiBeacon testClass1 testClass2"
  data-test-subj="test subject string"
  style="height:12px;width:12px"
/>
`;
